taskgroup/tasks.py

```diff
@@ -1,48 +1,56 @@
 import asyncio
 import collections.abc
+from typing import TYPE_CHECKING, Any, Generic, Type, TypeVar, cast
 
 
-@collections.abc.Coroutine.register
-class _Interceptor:
-    def __init__(self, coro, context):
-        self.__coro = coro
-        self.__context = context
+_YieldT = TypeVar("_YieldT")
+_SendT = TypeVar("_SendT")
+_ReturnT = TypeVar("_ReturnT", covariant=True)
 
-    def send(self, v):
-        return self.__context.run(self.__coro.send, v)
+if TYPE_CHECKING:
+    _Interceptor = collections.abc.Coroutine
+else:
+    @collections.abc.Coroutine.register
+    class _Interceptor:
```

Comment:
We probably want this to be `class _Interceptor(collections.abc.Coroutine[...]): ` instead of having if TYPE_CHECKING

Reply:
Dunno why I used register here